README.md

```diff
@@ -119,49 +123,52 @@ docker exec -it pinot-controller ./bin/pinot-admin.sh \
     -exec
 ```
 
-Once added, the OFFLINE table will not have any data.
-Let's add data in the next step.
-
+The OFFLINE table has no data. Let's add data in the next step.
 
-### Step 4: Load Data into the Movies Table
+#### Step 4: Load data into the movies table
 
-Use the following command to load data into the OFFLINE movies table:
+To load data into the OFFLINE table, run the following:
 
 ```bash
 docker exec -it pinot-controller ./bin/pinot-admin.sh \
     LaunchDataIngestionJob \
     -jobSpecFile /tmp/pinot/table/jobspec.yaml
 ```
+Now, both the REALTIME and OFFLINE tables are queryable, and you're ready to view, join, and query data in Pinot.
 
-Now, both the REALTIME and OFFLINE tables are queryable.
+## View, join, and query data in Pinot
 
-### Step 5: Apache Pinot Advanced Usage
+1. Open the [Pinot console](http://localhost:9000/#/query). 
+2. Click the **movies** and **movie_ratings** links to view data stored in each table. 
+3. To join the two datasets, do the following:
+   - Select the `Use Multi-Stage Engine` check box.
+   - Enter the following query under `SQL Editor`:
 
-To perform complex queries such as joins, open the Pinot console [here](http://localhost:9000/#/query) and enable `Use Multi-Stage Engine`. Example query:
-
-```sql
-select
-    r.rating latest_rating,
-    m.rating initial_rating,
-    m.title,
-    m.genres,
-    m.releaseYear
-from movies m
-         left join movie_ratings r on m.movieId = r.movieId
-where r.rating > .9
-order by r.rating desc
-    limit 10
-```
+        ```sql
+        select 
+            r.rating latest_rating, 
+            m.rating initial_rating, 
+            m.title, 
+            m.genres, 
+            m.releaseYear 
+        from movies m
+        left join movie_ratings r on m.movieId = r.movieId
+        where r.rating > .9
+        order by r.rating desc
+        limit 10
 
+        ```
 
-![alt](./images/results.png)
+4. Click `RUN QUERY`.
 
+    ![alt](./images/results.png)
```

Comment:
think running `make` yesterday gave me different results than were showing here (maybe not), but the important part was it was working for me

i just went back to check just now (after running`docker compose down` to clean up), ran `make` again, 
and get this error about invalid TableConfigs: "movies" (in the PInot console, movie_ratings does show up):
 

```
Waiting for Pinot Controller to be ready...
🍷 Pinot Controller is ready.
Waiting for Pinot Broker to be ready...
🍷 Pinot Broker is ready to receive queries.
🪲 Waiting for Kafka to be ready...
Connection to localhost port 9092 [tcp/XmlIpcRegSvc] succeeded!
🪲 Kafka is ready.
docker exec pinot-controller ./bin/pinot-admin.sh \
		AddTable \
		-tableConfigFile /tmp/pinot/table/movies.table.json \
		-schemaFile /tmp/pinot/table/movies.schema.json \
		-exec
[0.000s][warning][gc] -Xloggc is deprecated. Will use -Xlog:gc:gc-pinot-controller.log instead.
2024/04/17 00:05:31.017 INFO [AddTableCommand] [main] Executing command: AddTable -tableConfigFile /tmp/pinot/table/movies.table.json -offlineTableConfigFile null -realtimeTableConfigFile null -schemaFile /tmp/pinot/table/movies.schema.json -controllerProtocol http -controllerHost 172.19.0.5 -controllerPort 9000 -user null -password [hidden] -exec
2024/04/17 00:05:31.251 INFO [AddTableCommand] [main] {"code":400,"error":"Invalid TableConfigs: movies"}
sleep 5
docker exec pinot-controller ./bin/pinot-admin.sh \
		AddTable \
		-tableConfigFile /tmp/pinot/table/ratings.table.json \
		-schemaFile /tmp/pinot/table/ratings.schema.json \
		-exec
[0.000s][warning][gc] -Xloggc is deprecated. Will use -Xlog:gc:gc-pinot-controller.log instead.
2024/04/17 00:05:37.054 INFO [AddTableCommand] [main] Executing command: AddTable -tableConfigFile /tmp/pinot/table/ratings.table.json -offlineTableConfigFile null -realtimeTableConfigFile null -schemaFile /tmp/pinot/table/ratings.schema.json -controllerProtocol http -controllerHost 172.19.0.5 -controllerPort 9000 -user null -password [hidden] -exec
2024/04/17 00:05:37.519 INFO [AddTableCommand] [main] {"unrecognizedProperties":{},"status":"TableConfigs movie_ratings successfully added"}
docker exec pinot-controller ./bin/pinot-admin.sh \
		LaunchDataIngestionJob \
		-jobSpecFile /tmp/pinot/table/jobspec.yaml
[0.000s][warning][gc] -Xloggc is deprecated. Will use -Xlog:gc:gc-pinot-controller.log instead.
2024/04/17 00:05:38.590 ERROR [LaunchDataIngestionJobCommand] [main] Got exception to kick off standalone data ingestion job -
java.lang.RuntimeException: Failed to read from Schema URI - 'http://localhost:9000/tables/movies/schema'
	at org.apache.pinot.common.segment.generation.SegmentGenerationUtils.getSchema(SegmentGenerationUtils.java:94) ~[pinot-all-1.1.0-jar-with-dependencies.jar:1.1.0-c2606742bbc4b15cff857eb0ffe7ec878ff181bb]
	at org.apache.pinot.plugin.ingestion.batch.standalone.SegmentGenerationJobRunner.init(SegmentGenerationJobRunner.java:146) ~[pinot-batch-ingestion-standalone-1.1.0-shaded.jar:1.1.0-c2606742bbc4b15cff857eb0ffe7ec878ff181bb]
	at org.apache.pinot.spi.ingestion.batch.IngestionJobLauncher.kickoffIngestionJob(IngestionJobLauncher.java:148) ~[pinot-all-1.1.0-jar-with-dependencies.jar:1.1.0-c2606742bbc4b15cff857eb0ffe7ec878ff181bb]
	at org.apache.pinot.spi.ingestion.batch.IngestionJobLauncher.runIngestionJob(IngestionJobLauncher.java:121) ~[pinot-all-1.1.0-jar-with-dependencies.jar:1.1.0-c2606742bbc4b15cff857eb0ffe7ec878ff181bb]
	at org.apache.pinot.tools.admin.command.LaunchDataIngestionJobCommand.execute(LaunchDataIngestionJobCommand.java:132) [pinot-all-1.1.0-jar-with-dependencies.jar:1.1.0-c2606742bbc4b15cff857eb0ffe7ec878ff181bb]
	at org.apache.pinot.tools.Command.call(Command.java:33) [pinot-all-1.1.0-jar-with-dependencies.jar:1.1.0-c2606742bbc4b15cff857eb0ffe7ec878ff181bb]
	at org.apache.pinot.tools.Command.call(Command.java:29) [pinot-all-1.1.0-jar-with-dependencies.jar:1.1.0-c2606742bbc4b15cff857eb0ffe7ec878ff181bb]
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953) [pinot-all-1.1.0-jar-with-dependencies.jar:1.1.0-c2606742bbc4b15cff857eb0ffe7ec878ff181bb]
	at picocli.CommandLine.access$1300(CommandLine.java:145) [pinot-all-1.1.0-jar-with-dependencies.jar:1.1.0-c2606742bbc4b15cff857eb0ffe7ec878ff181bb]
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352) [pinot-all-1.1.0-jar-with-dependencies.jar:1.1.0-c2606742bbc4b15cff857eb0ffe7ec878ff181bb]
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346) [pinot-all-1.1.0-jar-with-dependencies.jar:1.1.0-c2606742bbc4b15cff857eb0ffe7ec878ff181bb]
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2311) [pinot-all-1.1.0-jar-with-dependencies.jar:1.1.0-c2606742bbc4b15cff857eb0ffe7ec878ff181bb]
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179) [pinot-all-1.1.0-jar-with-dependencies.jar:1.1.0-c2606742bbc4b15cff857eb0ffe7ec878ff181bb]
	at picocli.CommandLine.execute(CommandLine.java:2078) [pinot-all-1.1.0-jar-with-dependencies.jar:1.1.0-c2606742bbc4b15cff857eb0ffe7ec878ff181bb]
	at org.apache.pinot.tools.admin.PinotAdministrator.execute(PinotAdministrator.java:171) [pinot-all-1.1.0-jar-with-dependencies.jar:1.1.0-c2606742bbc4b15cff857eb0ffe7ec878ff181bb]
	at org.apache.pinot.tools.admin.PinotAdministrator.main(PinotAdministrator.java:202) [pinot-all-1.1.0-jar-with-dependencies.jar:1.1.0-c2606742bbc4b15cff857eb0ffe7ec878ff181bb]
Caused by: java.io.FileNotFoundException: http://localhost:9000/tables/movies/schema
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1994) ~[?:?]
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1599) ~[?:?]
	at org.apache.pinot.common.segment.generation.SegmentGenerationUtils.fetchUrl(SegmentGenerationUtils.java:239) ~[pinot-all-1.1.0-jar-with-dependencies.jar:1.1.0-c2606742bbc4b15cff857eb0ffe7ec878ff181bb]
	at org.apache.pinot.common.segment.generation.SegmentGenerationUtils.getSchema(SegmentGenerationUtils.java:92) ~[pinot-all-1.1.0-jar-with-dependencies.jar:1.1.0-c2606742bbc4b15cff857eb0ffe7ec878ff181bb]
	... 15 more
java.lang.RuntimeException: Failed to read from Schema URI - 'http://localhost:9000/tables/movies/schema'
	at org.apache.pinot.common.segment.generation.SegmentGenerationUtils.getSchema(SegmentGenerationUtils.java:94)
	at org.apache.pinot.plugin.ingestion.batch.standalone.SegmentGenerationJobRunner.init(SegmentGenerationJobRunner.java:146)
	at org.apache.pinot.spi.ingestion.batch.IngestionJobLauncher.kickoffIngestionJob(IngestionJobLauncher.java:148)
	at org.apache.pinot.spi.ingestion.batch.IngestionJobLauncher.runIngestionJob(IngestionJobLauncher.java:121)
	at org.apache.pinot.tools.admin.command.LaunchDataIngestionJobCommand.execute(LaunchDataIngestionJobCommand.java:132)
	at org.apache.pinot.tools.Command.call(Command.java:33)
	at org.apache.pinot.tools.Command.call(Command.java:29)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2311)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at org.apache.pinot.tools.admin.PinotAdministrator.execute(PinotAdministrator.java:171)
	at org.apache.pinot.tools.admin.PinotAdministrator.main(PinotAdministrator.java:202)
Caused by: java.io.FileNotFoundException: http://localhost:9000/tables/movies/schema
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1994)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1599)
	at org.apache.pinot.common.segment.generation.SegmentGenerationUtils.fetchUrl(SegmentGenerationUtils.java:239)
	at org.apache.pinot.common.segment.generation.SegmentGenerationUtils.getSchema(SegmentGenerationUtils.java:92)
	... 15 more
```

Reply:
Probably need to wait to check that the table is there when you search for it via REST API. Need to loop validator until table is found.